tests/ui/traits/next-solver/coroutine-higher-ranked-rigid.rs

```diff
@@ -0,0 +1,33 @@
+//@ compile-flags: -Znext-solver
+//@ edition: 2024
+//@ check-pass
```

Comment:
to my knowledge there was no existing ui test for this.

Please move this test to ui/generators or what not and also enable it for the old solver.